src/pybind/mgr/rbd_support/mirror_snapshot_schedule.py

```diff
@@ -364,6 +365,56 @@ def run(self) -> None:
             self.log.fatal("Fatal runtime error: {}\n{}".format(
                 ex, traceback.format_exc()))
 
+    def get_health_checks(self) -> Dict[str, Any]:
+        """
+        Returns a dictionary of health checks based on the synchronization status of images.
+        """
+        unsynced_images = []
+        try:
+            for pool_id, namespaces in self.images.items():
+                with self.module.rados.open_ioctx2(int(pool_id)) as ioctx:
+                    for namespace, images in namespaces.items():
+                        schedules = self.schedules.to_list(LevelSpec.make_global())
+                        for image_id, image_name in images.items():
+                            image_key = f"{pool_id}/{namespace}/{image_id}"
+                            if image_key not in schedules:
+                                continue
+                            image_schedule = schedules[image_key]['schedule']
+                            if not image_schedule:
+                                continue
+                            schedule_interval = image_schedule[0].get('interval')
+                            mirror_images = rbd.RBD().mirror_image_status_list(ioctx)
```

Comment:
How was this tested?

I'm asking because image status is updated by rbd-mirror daemon on the opposite cluster.  If there is a network issue or the daemon hangs in a particular manner at the time the image is caught up (i.e. when `remote_snapshot_timestamp` is equal to `local_snapshot_timestamp`), `mirror_image_status_list` here would continue returning two equal timestamps until rbd-mirror daemon is brought back to life...

Reply:
Alerts (I assume you are referring to prometheus alertmanager?) are separate from health checks (`ceph health [detail]`).  The former is an external component that may not even be configured, the latter is always there in any Ceph cluster.  Having a health check that doesn't fire in all cases it's supposed to fire is worse than having no health check at all: 

> "A false sense of security is worse than no security at all."

Reply:
> Alerts (I assume you are referring to prometheus alertmanager?) 

No. The rbd mirror daemon health alarm will be raised if we do not have any data from the daemon in the last 5 minutes, it does not depend of the prometheus manager module.

Reply:
> The rbd mirror daemon health alarm will be raised if we do not have any data from the daemon in the last 5 minutes, it does not depend of the prometheus manager module.

I'm not following what you mean by "rbd mirror daemon health alarm", but I'm pretty certain that it's not a health check and that it wouldn't show up in `ceph health [detail]` output.

Can you point me at the code that generates it?

Reply:
@idryomov , we have the cepoh health check "RECENT_CRASH"

Reply:
> Two issues:
> 
>     * `RECENT_CRASH` would be raised only on the cluster where rbd-mirror daemon crashes.  If it happens to be the "secondary" cluster (which is where rbd-mirror daemon is actually doing work), the user looking only at the "primary" cluster health wouldn't notice anything.  (I took "primary" and "secondary" in quotes because it's possible for the cluster to be primary for some subset images and secondary for another subset of images.)
> 
>     * Looking back at the bugs that we encountered in rbd-mirror daemon, it actually crashed in only a couple of cases.  Far more often it just locked up in various subtle ways.
> 
> 
> > Apart of that, I think that to have in health checks this kind of alert is useful (much more useful than to have nothing).
> 
> I disagree. If a particular health check doesn't exist, one would know that they need to set up their own monitoring and that they are responsible for that. If a health check does exist, one is likely to just rely on it -- and if it's broken in the sense that it isn't raised in all cases it's supposed to be raised, it's _very_ bad. Again:
> 
> > "A false sense of security is worse than no security at all."